### DISCUSSION TITLE
No colors in `ls`, etc.

### DISCUSSION THREAD

**Discussion by Eugeny** [Eugeny/tabby#5258] [FAQ & Tips]
## What is happening?

Color themes can only change colors of colored text that is printed to the terminal, not colorize text that is originally black and white. 

## How to make sure that a color scheme works

You can print a test table with various colours using `msgcat --color=test` on UNIX, or the following bash/zsh oneliner if `msgcat` isn't installed:

```bash
for x in {0..8}; do for i in {30..37}; do for a in {40..47}; do echo -ne "\e[$x;$i;$a""m\\\e[$x;$i;$a""m\e[0;37;40m "; done; echo; done; done; echo ""
```

A similar test script for PowerShell:

```powershell
$colors = [enum]::GetValues([System.ConsoleColor])
Foreach ($bgcolor in $colors){
    Foreach ($fgcolor in $colors) { Write-Host "$fgcolor|"  -ForegroundColor $fgcolor -BackgroundColor $bgcolor -NoNewLine }
    Write-Host " on $bgcolor"
}
```

### on macOS

By default, `ls` output on macOS is black and white.

You can re-enable output colorization by adding `export CLICOLOR=1` to the end of your `~/.zshrc` file (permanently) or by running `export CLICOLOR=1` in your shell (temporarily).

### on Linux

Alias `ls` to `ls --color=auto`, e.g:

```bash
alias ls='ls --color=auto'
```

**Comment by kabeza** [comment]
Hi
I'm having the same problem in ElementaryOS Hera (Ubuntu 20.04 based)
I set a color scheme in Tabby but it does not take effect. It only changes background/font color

When I run `ls --color=auto` the colors show correctly as set in color scheme

How could I set that alias permanent to be run at startup and tabby uses it by default in all tabs?

Thanks

**Reply by Eugeny** [reply]
See the end of the post

**Comment by AVGVSTVS96** [comment]
I don't see colors anywhere on MacOS Ventura, I do see color in ls when running ls --color but my username is not colored as it should be according the the color schemes I'm using, I have read similar posts but have not actually found an answer as to how to solve this. Can you provide some insight on how to get color working? I'm using the default shell ZSH but also tried Bash and it's not colored either. 

I don't see any of the styling provided by the color schemes at all, everything is completely plain text :(

**Reply by Eugeny** [reply]
Please read the post, a terminal doesn't make text that is white on black (your prompt including the username) colored, it only changes the colors that are already there. Grab a turnkey prompt customizer like https://ohmyz.sh/ to get a nicer prompt.

**Reply by RecklessRaccoon** [reply]
Hi, a bit too late...is it possible that tabby would show the color outputs as MobaXterm does, for eg, showing colored outputs for IP addresses, and interface errors, any warnings, etc. For me, even though I have selected colored themes, it still shows plain text only. Can you advise how I could achieve that on Tabby, similar to MobaXterm? Thanks!

**Reply by Eugeny** [reply]
@RecklessRaccoon consider the `highlight` plugin

**Reply by RecklessRaccoon** [reply]
I installed the highlight plugin and restarted the terminal, but still no luck. The output is still plain white text...

**Comment by Kisaragi-ng** [comment]
I stumbled upon this discussion because _somehow_ my terminal color doesn't work even when using `ls='ls --color=auto'`
turns out i have set my "Minimum contrast ratio" in Settings->Appearance way too high (above 20), got it fixed now.

**Comment by hitzhangjie** [comment]
This doesn't solve the problem at all. Please don't close relevant issues too quickly. We should fix this kind of issue in a more friendly way. 

In my case:
```
alias ls='ls --color -vp'
alias ll='ls --color -alvp'
```

This was set, but when I switching color scheme in Tabby, it doesn't change the color at all:
<img width="767" alt="image" src="https://github.com/user-attachments/assets/203f7a7c-b22e-403f-8c0c-6fec05a3779d" />

And this:
```
for x in {0..8}; do for i in {30..37}; do for a in {40..47}; do echo -ne "\e[$x;$i;$a""m\\\e[$x;$i;$a""m\e[0;37;40m "; done; echo; done; done; echo ""
```

it just output things like this:
<img width="1490" alt="image" src="https://github.com/user-attachments/assets/7990226a-cadf-4fd3-911f-80985f430399" />


So I think the output color doesn't conforms to the specification, actually on darwin and linux platform, they use different form to represent colors:
<img width="355" alt="image" src="https://github.com/user-attachments/assets/dc84d733-9ce1-4238-99fd-aed7b9efe1f5" />

This is some shell code I wrote before.

**Reply by hitzhangjie** [reply]
Well, I find change `minimum contrast' to smaller value like 1, it looks better.

<img width="659" alt="image" src="https://github.com/user-attachments/assets/8c60ddfa-e656-475d-8545-3d1004281b47" />

But I don't know if it works normally or not, because the color theme shown is misunderstanding, though I don't have better solution to display them.

**Reply by gddg** [reply]
I changed the minimum contrast 50, and it won't render the font color. This is the problem.
tabby-1.0.223-linux-x64
